Tribler/Policies/BoostingManager.py

```diff
@@ -0,0 +1,676 @@
+                self.stop_download(torrent)
+            for torrent in torrents_start:
+                self.start_download(torrent)
+
+            self._logger.info("Selecting from %s torrents %s start download", len(torrents), len(torrents_start))
+
+        self.session.lm.threadpool.add_task(self._select_torrent, self.settings.swarm_interval)
+
+    def load_config(self):
+        """
+        load config in file configuration and apply it to manager
+        """
+        config = ConfigParser.ConfigParser()
+        config.read(self.settings.config_file)
+        validate_source = lambda s: unhexlify(s) if len(s) == 40 and not s.startswith("http") else s
+
+        def _add_sources(value):
+            """
+            adding sources in configuration file
+            """
+            for boosting_source in json.loads(value):
+                boosting_source = validate_source(boosting_source)
+                self.add_source(boosting_source)
+
+        def _archive_sources(value):
+            """
+            setting archive to sources
+            """
+            for archive_source in json.loads(value):
+                archive_source = validate_source(archive_source)
+                self.set_archive(archive_source, True)
+
+        def _set_enable_boosting(value, enabled):
+            """
+            set disable/enable source
+            """
+            for boosting_source in json.loads(value):
+                boosting_source = validate_source(boosting_source)
+                if not self.boosting_sources[boosting_source]:
+                    self.add_source(boosting_source)
+                self.boosting_sources[boosting_source].enabled = enabled
+
+        switch = {
+            "boosting_sources": {
+                "cmd": _add_sources,
+                "args": (None,)
+            },
+            "archive_sources": {
+                "cmd": _archive_sources,
+                "args": (None,)
+            },
+            "boosting_enabled": {
+                "cmd": _set_enable_boosting,
+                "args": (None, True)
+            },
+            "boosting_disabled": {
+                "cmd": _set_enable_boosting,
+                "args": (None, False)
+            },
+        }
+
+        switch_policy = {
+            "random" : RandomPolicy,
+            "creation" : CreationDatePolicy,
+            "seederratio" : SeederRatioPolicy
+        }
+
+        for k, val in config.items(__name__):
+            try:
+                if k in self._saved_attributes:
+                    object.__setattr__(self.settings, k, int(val))
+                elif k == "policy":
+                    self.settings.policy = switch_policy[val](self.session)
+                else:
+                    switch[k]["cmd"](*((switch[k]['args'][0] or val,) + switch[k]['args'][1:]))
+            except KeyError:
+                self._logger.error("Key %s can't be applied", k)
+
+    def save_config(self):
```

Comment:
I think we should think about a new configuration approach. This method is full with work-arounds to get the data in the right form :/